### PR TITLE
Stabilize telemetry consent and promote telemetry to production

### PR DESCRIPTION
**Pull request by RamonArjona4**
## 📖 Description

Promotes MXC telemetry from experimental configuration to the stable top-level
`telemetry` section and completes the user-facing privacy contract across the
EXE, Rust SDK, C ABI, C#, and Node.js.

Collection remains Windows-only and requires every term below at event-write
time:

`per_run_requested && current_version_consent_granted && admin_policy_allows && provider_available`

Key changes:

- Adds one Rust-owned, versioned `en-US` consent resource that every EXE/SDK
  presenter must render verbatim, with future localization fallback boundaries.
- Replaces direct grant APIs with presenter-bound typed decisions. No request,
  No, dismissal, presenter failure, malformed state, or legacy/unknown prompt
  version authorizes collection.
- Adds typed JSON maintenance actions (`request`, `withdraw`, `status`) through
  the existing config loader and a separate generated maintenance schema.
- Keeps `--telemetry-consent-status` read-only; old grant/revoke/source flags
  are non-mutating migration tombstones.
- Exposes sync/async presenter, status, withdrawal, and stable per-run telemetry
  enablement through the public Rust, C#, and Node SDK surfaces.
- Treats administrative policy as a deny-only ceiling; it can block but never
  opt a user in. Withdrawal remains available while policy is blocked.
- Rechecks consent and policy immediately before every event write so a
  withdrawal or new policy block stops later emission from long-lived
  processes.
- Preserves the established ETW identities `MXC.Execution` and `MXC.Error`.
- Adds generated contract/parity gates, named release consent/policy tests, and
  isolated debug consent/ETW smoke tests to CI.
- Documents the exact collected/excluded fields and adds a privacy/legal review
  package. Approval remains a release gate.

Legacy `experimental.telemetry` is rejected with an actionable migration error.

## 🔗 References

- Related: #492
- Follow-up context: #690, #691
- Design: `docs/telemetry/telemetry-consent-design.md`
- Privacy review package:
  `docs/telemetry/telemetry-consent-privacy-review.md`

## 🔍 Validation

- `cargo fmt --all -- --check`
- `cargo clippy --workspace --all-targets -- -D warnings`
- `cargo test --workspace`
- `cargo test --release -p wxc_common telemetry::consent`
- `cargo test --release -p wxc_common telemetry::policy`
- Node SDK build and full unit suite
- .NET SDK: 63 tests passed
- Telemetry consent/schema/SDK codegen checks
- Telemetry policy, C# binding, and error-code parity checks
- Isolated debug telemetry consent smoke test
- ETW smoke script syntax validated locally; elevated ETW execution is wired
  into the Windows x64 CI job
- Independent stabilization review completed; its confirmed Node timeout/EPIPE
  finding was fixed and the affected suite rerun

## ✅ Checklist

- [x] Signed the [Contributor License Agreement](https://cla.opensource.microsoft.com)
- [x] Linked to an issue
- [x] Updated documentation (if applicable)
- [x] Updated [Copilot instructions](.github/copilot-instructions.md) (if build, architecture, or conventions changed)
- [x] If this PR changes `Cargo.lock`, the `dependency-feed-check` check passes (see [docs/pull-requests.md](https://github.com/microsoft/mxc/blob/main/docs/pull-requests.md))

## 📋 Issue Type

- [ ] Bug fix
- [x] Feature
- [ ] Task

---

GitHub Actions runs the PR validation build automatically. The ADO pipeline
(`MXC-PR-Build`) is the Azure version of the PR pipeline, kept in parity with
the GitHub Actions build; it runs on merge to `main`, and Microsoft reviewers
with write access can trigger it on a PR with `/azp run`. See
[docs/pull-requests.md](https://github.com/microsoft/mxc/blob/main/docs/pull-requests.md).

If the `dependency-feed-check` check fails on a new dependency, the crate must
be added to the feed before the PR can pass. See
[docs/pull-requests.md](https://github.com/microsoft/mxc/blob/main/docs/pull-requests.md)
for the steps.

### CODE REVIEW

File: src/ffi/mxc_ffi/src/lib.rs [hunk 1]
```diff
@@ -41,13 +41,40 @@
...
+fn report_panic(operation: &str, payload: &(dyn Any + Send)) {
+    let message = payload
+        .downcast_ref::<&str>()
+        .copied()
+        .or_else(|| payload.downcast_ref::<String>().map(String::as_str))
+        .unwrap_or("<non-string panic payload>");
+    eprintln!(
+        "mxc: internal error: panic caught at the FFI boundary in {operation}: {message}. \
+         Failing closed; the calling process is unaffected."
+    );
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply src/ffi/mxc_ffi/src/lib.rs]: Fixed: `report_panic` now goes through a `report_to_stderr` helper that locks stderr and discards the `io::Result`. Also fixed the consent-write-failure arm, which sits *outside* `catch_unwind` — a panic there would have unwound into foreign frames. Corrected the doc comment that claimed writing to a closed stderr is a no-op.

File: tests/scripts/run_telemetry_consent_smoke_test.ps1 [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+$ConsentFile = Join-Path $TempDir "mxc\telemetry-consent.json"
+
+try {
+    Write-Host "Running telemetry consent CLI smoke test..." -ForegroundColor Cyan
+
+    $status0 = & $WxcExec --telemetry-consent-status
+    Assert-Consent $status0 "undetermined" "true" "fresh store status"
+
+    $grant = & $WxcExec --telemetry-consent-grant --telemetry-consent-source prompt
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply tests/scripts/run_telemetry_consent_smoke_test.ps1]: Fixed: the smoke test now proves isolation read-only *before* any write, by seeding two different consent records (`granted`, then `denied`) and requiring the CLI to report each back. A binary ignoring the override reads one fixed store and cannot match both.

File: src/ffi/mxc_ffi/src/lib.rs [hunk 3]
```diff
@@ -405,4 +608,254 @@ mod tests {
...
+    //
+    // These tests mutate `MXC_TEST_LOCALAPPDATA_OVERRIDE`, a debug-build-only
+    // escape hatch `wxc_common::telemetry::consent` reads instead of trusting
+    // the real `LOCALAPPDATA` env var (see that module for the security
+    // rationale — a release binary compiles this override out entirely).
+    // Every test that touches consent state must serialize on
+    // `CONSENT_ENV_LOCK` and restore the original value afterward — this
+    // process may also run the `mxc_run` tests above in parallel, none of
+    // which touch this override, but two consent tests racing each other
+    // would otherwise stomp on each other's redirected directory.
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply src/ffi/mxc_ffi/src/lib.rs]: Fixed: `LocalAppDataGuard` became `TelemetryTestEnv` and now owns a `PolicyKeyGuard` (and therefore `POLICY_LOCK`) in a fixed consent-then-policy order, closing the consent-vs-policy race and ambient-machine-policy leakage. The whole section is now `#[cfg(debug_assertions)]` so a release-build run can never read or write the developer's real consent store.

File: src/core/wxc_common/src/telemetry/consent_cli.rs [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+        }
+    }
+
+    // --telemetry-consent-status (or the post-grant/-revoke confirmation)
+    // always prints the resulting state so callers get a single, uniform
+    // JSON contract regardless of which flag was passed.
+    let response = ConsentStatusResponse {
+        consent: consent::get_consent().as_str(),
+        needs_prompt: consent::needs_consent_prompt(),
+        policy: policy::get_policy().as_str(),
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply src/core/wxc_common/src/telemetry/consent_cli.rs]: Fixed: the status JSON is now a single coherent snapshot — one `get_consent()` and one `get_policy()`, with `needs_prompt` derived as `policy.allows_collection() && consent.needs_prompt()` (algebraically equivalent to `needs_consent_prompt()`).

File: sdk/node/src/telemetry.ts [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+    return {
+      state: 'undetermined',
+      needsPrompt: false,
+      policy: 'blocked',
+      error: `failed to read telemetry consent: ${detail}`,
+    };
+  }
+  const { state, needsPrompt, policy } = parseConsentOutput(stdout);
+  if (state === 'undetermined' && !stdout.includes('"undetermined"')) {
+    const detail = `unrecognised telemetry consent output: ${stdout.trim().slice(0, 200)}`;
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply sdk/node/src/telemetry.ts]: Fixed: removed the `stdout.includes('"undetermined"')` substring heuristic. `ParsedConsentOutput` gained an optional `error`, and policy-parse failure is now reported separately from consent-parse failure.

File: sdk/dotnet/Microsoft.Mxc.Sdk/MxcTelemetry.cs [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+
+/// <summary>
+/// Administers MXC's persisted, MXC-owned telemetry consent flag. See
+/// docs/telemetry/telemetry-consent-design.md for the full design.
+///
+/// This is deliberately UI-agnostic: it does not render a consent prompt.
+/// A hosting application should call <see cref="GetConsent"/> once (e.g.
+/// before its first sandbox run) and, if it returns
+/// <see cref="TelemetryConsentState.Undetermined"/>, show its own consent UI
+/// and then call <see cref="SetConsent"/> with the user's choice. A settings
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply sdk/dotnet/Microsoft.Mxc.Sdk/MxcTelemetry.cs]: Fixed: the class documentation now directs hosts to `NeedsConsentPrompt()` rather than branching on `GetConsent()`, which ignores administrative policy.

File: sdk/dotnet/Microsoft.Mxc.Sdk/MxcTelemetry.cs [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+/// page can call either method again at any later time to let the user
+/// change their mind.
+/// </summary>
+public static class MxcTelemetry
+{
+    static MxcTelemetry()
+    {
+        NativeLibraryResolver.Initialize();
+    }
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply sdk/dotnet/Microsoft.Mxc.Sdk/MxcTelemetry.cs:33]: Fixed: `NativeLibraryResolver.Initialize()` wraps `SetDllImportResolver` in try/catch so a throwing static constructor can no longer poison the type. The diagnostic write itself is now nested in its own try/catch — a host that replaced or closed `Console.Error` would otherwise re-poison the ctor this catch exists to protect.

File: src/core/mxc-sdk/src/lib.rs [hunk 8]
```diff
@@ -67,6 +67,74 @@
...
+/// The policy is a ceiling, never a grant: an administrator who permits
+/// telemetry has not consented on the user's behalf, so an explicit user grant
+/// is still required. A blocking policy also makes [`needs_consent_prompt`]
+/// return `false`, so a host following the pattern above will not ask the user
+/// to decide something MXC would then ignore.
+pub mod telemetry {
+    pub use wxc_common::telemetry::consent::{
+        get_consent, needs_consent_prompt, set_consent, ConsentState,
+    };
+    pub use wxc_common::telemetry::policy::{get_policy, is_blocked_by_policy, PolicyState};
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply src/core/mxc-sdk/src/lib.rs]: Fixed: the facade moved into a dedicated `src/core/mxc-sdk/src/telemetry.rs` with SDK-owned `ConsentState`/`PolicyState` enums, `From` conversions, delegating predicates and tests. `lib.rs` is now just `pub mod telemetry;`. The README no longer calls it a pure re-export.

File: docs/telemetry/telemetry-consent-design.md [hunk 9]
```diff
@@ -0,0 +1,642 @@
+# Telemetry Consent — Feature Spec
+
+> Status: **Proposal / design plan** — no code has been written yet. This
+> follows the "Write a Feature Spec" step in
+> [`docs/authoring-a-new-feature.md`](../authoring-a-new-feature.md) so the
+> team can align before implementation begins.
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply docs/telemetry/telemetry-consent-design.md]: Fixed: the header now reads "Implemented" instead of "Proposal / no code written yet", and the cross-reference points at §9.

File: .github/copilot-instructions.md [hunk 10]
```diff
@@ -226,6 +229,18 @@ The parser deserializes JSON directly into the typed wire model (`wxc_common::wi
...
+
+- **Everything fails closed.** Any error, unreadable value, corrupt file, missing native library, or ambiguity must resolve to "no telemetry" (`Undetermined` consent / `Blocked` policy), never to a permissive state.
+- **Policy may restrict, but may never substitute for, consent.** The administrative policy is a deny-only ceiling: it can subtract from what a user permitted, never add to it. An administrator cannot opt a user in — a denied or never-asked user stays opted out even under `AllowTelemetry=3`. Keep the terms combined with `&&`; never add a policy value or config path that grants collection on its own.
+- **MXC owns its own consent state.** It must never read or infer from the Windows system telemetry consent. The consent store is a per-user JSON file; the policy is `HKLM\SOFTWARE\Policies\Mxc` → `AllowTelemetry` (`REG_DWORD`).
+- **One definition, distributed to the bindings.** The Rust `ConsentState` / `PolicyState` enums are the source of truth; the FFI, C#, and TypeScript layers marshal the same strings. `scripts/check-telemetry-policy-parity.js` is a CI gate that fails if the four `PolicyState` spellings drift apart — run it after touching any of the three.
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply .github/copilot-instructions.md]: Fixed: the instructions no longer describe the parity script as a CI gate while it is unwired.

File: sdk/dotnet/Microsoft.Mxc.Sdk/Native/NativeLibraryResolver.cs [hunk 11]
```diff
@@ -65,11 +103,23 @@ private static IEnumerable<string> CandidatePaths()
         yield return Path.Combine(baseDir, file);
         yield return Path.Combine(baseDir, "runtimes", RuntimeInformation.RuntimeIdentifier, "native", file);
 
-        // Dev layout: walk up looking for the Cargo target dir.
+        // Dev layout: walk up looking for the Cargo target dir. Probe the
+        // Cargo profile matching this assembly's build configuration first:
+        // a Release C# build binding against a stale debug mxc_ffi would pick
+        // up debug-only behaviour (e.g. the LOCALAPPDATA consent-store
+        // override), which is exactly what a Release build must not do.
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply sdk/dotnet/Microsoft.Mxc.Sdk/Native/NativeLibraryResolver.cs]: Fixed: the dev-layout Cargo-target-dir candidates (profile-specific) are now yielded before the generic `baseDir`/`runtimes/<rid>/native` fallbacks, rather than after. A Release build now always prefers a release-built `mxc_ffi` over any leftover debug DLL staged into `runtimes/<rid>/native` by a prior `build.bat --debug` run; the generic fallbacks are only reached once no Cargo target dir is found at all (e.g. a packaged NuGet consumer).

File: scripts/check-telemetry-policy-parity.js [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+const csharpDefault = /_\s*=>\s*TelemetryPolicyState\.Blocked/.test(
+  parseBody[1]
+);
+if (!csharpDefault) {
+  errors.push(
+    "C# ParsePolicyState must fail closed with `_ => TelemetryPolicyState.Blocked`"
+  );
+}
+// The default arm covers "blocked", so treat it as handled.
+csharpStates.add("blocked");
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply scripts/check-telemetry-policy-parity.js]: Fixed: the parity script now accepts the fail-closed default arm being either a direct `_ => TelemetryPolicyState.Blocked` or `_ => UnrecognizedPolicyState(value)`, and when the helper form is used it verifies `UnrecognizedPolicyState` itself returns `Blocked`. `node scripts/check-telemetry-policy-parity.js` now passes against the current C# code.

File: src/core/wxc_common/src/telemetry/consent.rs [hunk 13]
```diff
@@ -0,0 +1,835 @@
...
+        use windows::Win32::Security::TOKEN_QUERY;
+        use windows::Win32::System::Threading::{GetCurrentProcess, OpenProcessToken};
+
+        let mut token = HANDLE::default();
+        // SAFETY: `GetCurrentProcess` returns a pseudo-handle that needs no
+        // release, and `token` is a valid out-pointer for the duration of the
+        // call. On success the returned handle is immediately wrapped in
+        // `OwnedHandle`, whose `Drop` closes it exactly once.
+        unsafe { OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &mut token) }.ok()?;
+        Some(crate::process_util::OwnedHandle::new(token))
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply src/core/wxc_common/src/telemetry/consent.rs:504]: Fixed: `process_token()` now requests `TOKEN_QUERY | TOKEN_IMPERSONATE`. `SHGetKnownFolderPath` duplicates the handle into an impersonation token internally, so it requires both rights on systems that enforce that contract; a `TOKEN_QUERY`-only handle could make known-folder resolution fail, causing consent to always read as `Undetermined` and every grant/revoke to fail closed.

File: src/core/wxc/src/main.rs [hunk 14]
```diff
@@ -771,6 +830,13 @@ fn main() {
         Err(e) => eprintln!("DACL recovery failed: {e}"),
     }
 
+    // --telemetry-consent-{status,grant,revoke}: administer the persisted
+    // consent flag and exit. Run before --probe (cheapest possible fast
+    // path — no config parsing, no policy defaults needed).
+    if handle_telemetry_consent_flags(&cli) {
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply src/core/wxc/src/main.rs]: Fixed: `handle_telemetry_consent_flags(&cli)` now runs immediately after CLI parsing, before `--force-reclaim` env propagation and `recover_orphaned_state()` (DACL recovery) — matching the ordering already used by the Linux (`lxc-exec`) and macOS (`mxc-exec-mac`) executors. A read-only `--telemetry-consent-status` query (e.g. Node's 5-second-timeout consent getter) can no longer be delayed or time out behind unrelated recovery work, and a status read can no longer trigger a DACL mutation as a side effect.

**Comment by kanismohammed** [comment]
Could we move this feature from experimental to stable before it ships? Telemetry consent and the administrative policy ceiling form a user-facing privacy contract, so callers should not need `--experimental` or an experimental config block once the consent, fail-closed behavior, and policy semantics are ready.

As part of stabilization, please also expose per-run telemetry enablement through the public Rust SDK. The PR adds consent APIs, but `SandboxPolicy`/`SandboxRequest` currently has no telemetry setting, so in-process Rust callers can grant consent but cannot enable telemetry without invoking the standalone executor. The parity/codegen checks and release-build consent coverage should also be wired into CI as stable-feature exit criteria.

File: tests/scripts/run_telemetry_etw_smoke_test.ps1 [hunk 15]
```diff
@@ -132,13 +132,13 @@ Write-Host "ETW session started, writing to $etlFile"
...
-    # registered during init (before execution); the MXC.Execution / MXC.Error
+    # The provider is registered during init (before execution); the
+    # MXC.Execution / MXC.Error
     # events are emitted on completion, after the runner returns. The sandbox
     # itself may fail (e.g. AppContainer prerequisites), but completion
     # telemetry still fires for the failure, so events should be captured.
     $proc = Start-Process -FilePath $wxcExe `
-        -ArgumentList "--debug", "--experimental", $configFile `
+        -ArgumentList "--debug", $configFile `
         -PassThru -NoNewWindow -Wait
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply tests/scripts/run_telemetry_etw_smoke_test.ps1]: Addressed in de7a691. The ETW smoke test now refuses release binaries, creates unique consent and HKCU policy overrides, proves both redirects before any MXC write, obtains consent through the canonical session-bound JSON presenter flow, requires effective `granted` plus policy `allowed` before starting ETW, preserves the `MXC.Execution`/`MXC.Error` assertions, and restores the ETW session, environment, policy key, consent directory, and ETL files in `finally`. The shared consent smoke test uses the same isolation pattern and passes locally; the elevated ETW run is wired into the Windows x64 CI job.

**Comment by RamonArjona4** [comment]
Implemented the requested stabilization in de7a691:

- Telemetry is now stable top-level `telemetry`; legacy `experimental.telemetry` is rejected with migration guidance.
- Public Rust per-run enablement remains source-compatible as `SandboxRequest::set_telemetry_enabled(bool)` and is covered for both run and streaming paths.
- Consent is presenter-bound across Rust, C ABI, C#, Node, and the EXE using one versioned canonical `en-US` resource; no direct public grant path remains.
- Missing requests, No, dismissal, presenter failures, legacy grants, malformed state, and blocked/unreadable policy all fail closed. Withdrawal remains idempotent and available while blocked.
- Consent and policy are rechecked before every event emission.
- `MXC.Execution` and `MXC.Error` event identities are preserved and pinned by tests.
- Telemetry policy parity, maintenance codegen, .NET binding/error parity, named release consent/policy tests, and isolated consent/ETW smoke tests are wired into CI.
- The PR title/body now describe the stable contract and current validation instead of the former experimental proof-of-concept.

Validation is clean across workspace fmt/clippy/tests, release consent/policy tests, Node, .NET (63 tests), codegen/parity gates, and the isolated consent smoke test. The privacy/legal review package is included and records approval as a pending release gate.

**Comment by RamonArjona4** [comment]
Fixed the ETW smoke false failure in 84cd06a. The test had completed and printed PASS, but the final cleanup attempted to delete an already-stopped real-time trace session and leaked logman's expected exit code 1. The script now stops without the redundant delete, uses tracerpt's less-restricted XML mode, and explicitly exits 0 only after successful completion. The rerun passed, including the Windows x64 'Run isolated telemetry ETW smoke test' step: https://github.com/microsoft/mxc/actions/runs/31457596772

**Comment by RamonArjona4** [comment]
Added a normative SDK presenter contract in 9c33aae and linked it from the Rust, C#, and Node SDK READMEs. It now explicitly defines verbatim resource rendering, Yes/No mappings, close/cancel/timeout as Dismissed, UI failures as presenter errors, learn-more URL behavior, MXC-owned persistence, status/withdrawal expectations, and the AlreadyGranted/PolicyBlocked/NotApplicable no-presenter outcomes.

File: sdk/dotnet/Microsoft.Mxc.Sdk/MxcLifecycle.cs [hunk 16]
```diff
@@ -186,21 +197,35 @@ public static async Task<RunResult> ExecInSandboxAsync(
...
 
+    /// <summary>
+    /// Compatibility overload retaining the original positional cancellation
+    /// token signature.
+    /// </summary>
+    public static Task<RunResult> ExecInSandboxAsync(
+        SandboxId id,
+        string command,
+        CancellationToken cancellationToken) =>
+        ExecInSandboxAsync(id, command, null, cancellationToken);
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply sdk/dotnet/Microsoft.Mxc.Sdk/MxcLifecycle.cs]: Fixed in e60e156. `ExecInSandboxAsync` now preserves `CancellationToken` in its original third-parameter position and adds `StateAwareOperationOptions` as the optional fourth parameter, removing the overlapping overload. A compile-time regression test covers `ExecInSandboxAsync(id, command, default)`, and all 16 focused lifecycle tests pass.

File: tests/scripts/run_telemetry_etw_smoke_test.ps1 (hunk 15 above)
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply tests/scripts/run_telemetry_etw_smoke_test.ps1]: Addressed in de7a691 and validated by the passing Windows x64 ETW smoke workflow. The test now uses a debug binary with isolated consent and policy overrides, explicitly grants consent before capture, and restores the isolated state in cleanup.

**Comment by RamonArjona4** [comment]
Addressed the latest global review findings in 27a5577 and ec7cbc0: SDK telemetry now always releases its provider reference even when a live consent/policy recheck suppresses emission; stable telemetry rejects unknown fields and regenerated schema/types enforce that contract; collector-facing docs consistently preserve MXC.Execution/MXC.Error; the policy status example matches the typed maintenance response; CI documentation reflects the parity gate; SDK wording uses presenter-bound request/withdraw semantics; privacy approval is consistently documented as a release gate; and the C# `Undetermined` documentation now requires policy-aware `NeedsPrompt`/`NeedsConsentPrompt()` rather than prompting from consent state alone. Added regressions for provider release after authorization closes and telemetry field typos.

**Comment by RamonArjona4** [comment]
Addressed the two latest suppressed documentation findings in 67486b4. The blocked-policy status example now preserves storedState/ffectiveState as granted, reports 
eason: null, and represents the block only through the separate policy ceiling. The C# Undetermined guidance now references only the actual policy-aware API, MxcTelemetry.NeedsConsentPrompt().

**Comment by RamonArjona4** [comment]
Addressed the suppressed Node one-shot telemetry finding in 471df2d. options.telemetry is now copied into the shared one-shot execution config used by spawnSandbox, spawnSandboxAsync, and both spawnSandboxFromConfig modes. Omitting the option preserves config.telemetry; explicitly supplying it overrides the config without mutating the caller object. Added coverage for copy, preservation, and precedence; all 270 Node unit tests pass.

File: src/mxc_telemetry/src/lib.rs [hunk 17]
```diff
@@ -321,6 +341,8 @@ mod tests {
         let _ = init("0.0.0-test", "dev");
         let _ = init("0.0.0-test", "dev");
         shutdown();
+        assert!(is_active(), "one retained registration must remain active");
```
[the remark this answers is not in the document]
Comment (RamonArjona4) [reply src/mxc_telemetry/src/lib.rs]: Fixed in b40368d. The test now records whether `init()` succeeded, verifies the second call matches that result, asserts one retained registration only when initialization succeeded, and confirms the final release is inactive on every platform. All 15 `mxc_telemetry` tests pass.

**Comment by RamonArjona4** [comment]
This monolithic PR has been superseded by a dependency-ordered stack for focused human review: #818 (normative privacy/telemetry docs), #819 (consent and policy foundation), #820 (stable runtime and executors), #821 (Rust SDK and C ABI), and #822 (Node.js and .NET SDKs). The final stack tip is tree-identical to this PR. Please review and merge bottom-up starting with #818.

**Comment by RamonArjona4** [comment]
Closing as superseded by #818–#822; existing discussion is preserved here as historical context.
